File: code/stable/bmbnd/src/swift/OutputFormat.swift
/** OutputFormat.swift
    Provides the function for writing outputs
    - Authors: Jason Balaci
*/
import Foundation

/** Writes the output values to output.txt
    - Parameter y_B: deflection at a particular point along the beam (m)
*/
func write_output(_ y_B: (Double, Float) -> Double) throws -> Void {
    var outputfile: FileHandle
    do {
        outputfile = try FileHandle(forWritingTo: FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.appendingPathComponent("output.txt"))
    } catch {
        throw "Error opening file."
    }
    do {
        try outputfile.write(contentsOf: Data("y_B = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(y_B).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.close()
    } catch {
        throw "Error closing file."
    }
}

File: code/stable/bmbnd/src/cpp/OutputFormat.cpp
#include "OutputFormat.hpp"

#include <fstream>
#include <iostream>
#include <string>
#include <vector>

using std::ofstream;
using std::string;
using std::vector;

void write_output(vector<double> &y_B) {
    ofstream outputfile;
    outputfile.open("output.txt", std::fstream::out);
    outputfile << "y_B = ";
    outputfile << "[";
    for (int list_i1 = 0; list_i1 < (int)(y_B.size()) - 1; list_i1++) {
        outputfile << y_B.at(list_i1);
        outputfile << ", ";
    }
    if ((int)(y_B.size()) > 0) {
        outputfile << y_B.at((int)(y_B.size()) - 1);
    }
    outputfile << "]" << std::endl;
    outputfile.close();
}
